### PR TITLE
chore(*): get rid of Bower in favor of Yarn aliases & checked-in packages

### LINKED ISSUE #14961: Request dependency <=2.68 opens to potential memory exposure vulnerability
**Do you want to request a _feature_ or report a _bug_?**
This ticket is to report a a potential security vulnerability caused by the request dependency.

**What is the current behavior?**
Various of the dependencies used by angular.js make use of a vulnerable version of the request package (<2.68) that allow potential memory exposure.

Involved dependencies are: insight, fsevents

details:
- https://snyk.io/vuln/npm:request:20160119
- https://github.com/yeoman/insight/pull/48
- https://github.com/strongloop/fsevents/issues/145

In order to address a short term fix it is suggested to modify the current npm shrinkwrap to use request==2.74.0

### LINKED ISSUE #16268: Move from bower to yarn aliases
We are currently using bower because we need multiple version of jquery for testing. This is also possible with yarn aliases, which will streamline our workflow.

This will partially fix https://github.com/angular/angular.js/issues/14961

- [ ] the Contributing.md (soon: DEVELOPERS.md) must be updated after bower is removed

### PR DESCRIPTION
**Pull request by mgol**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**

Build change.

**What is the current behavior? (You can also link to an open issue here)**

Bower is used to install jQuery & Closure Compiler-related packages.

**What is the new behavior (if this is a feature change)?**

Most packages are installed from the npm registry, Yarn aliases are used to manage multiple jQuery versions. `ng-closure-compiler` was checked in to the repository.

**Does this PR introduce a breaking change?**

No.

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our [guidelines](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#commits)
- [x] Fix/Feature: [Docs](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#documentation) have been added/updated
- [x] Fix/Feature: ~~Tests have been added~~; existing tests pass

**Other information**:

Bower was used to install multiple versions of jQuery which is now handled
using Yarn aliases. The remaining two packages, closure-compiler and
ng-closure-compiler were installed from zip files which is not supported by Yarn
(see https://github.com/yarnpkg/yarn/issues/1483); the first of them exists
on npm as the google-closure-compiler but only versions newer than we used are
published and they don't work with ng-closure-compiler so - instead - both were
checked in to the repository.

Fixes #16268
Fixes #14961
Ref yarnpkg/yarn#1483

### CODE REVIEW

File: docs/config/services/getVersion.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ var path = require('canonical-path');
 /**
  * dgService getVersion
  * @description
- * Find the current version of the bower component (or node module)
+ * Find the current version of the node module
```
Comment (Narretz) [inline docs/config/services/getVersion.js]: If this function is only used to get node modules (afaict it does), we can simplify the implementation and remove the second and third arguments. I think this happened already when we moved most of the bower stuff to node.

File: DEVELOPERS.md [hunk 2]
```diff
@@ -483,4 +483,4 @@ You can see an example of a well-defined example [in the `ngRepeat` documentatio
 [karma-browserstack]: https://github.com/karma-runner/karma-browserstack-launcher
 [karma-saucelabs]: https://github.com/karma-runner/karma-sauce-launcher
 [unit-testing]: https://docs.angularjs.org/guide/unit-testing
-[yarn-install]: https://yarnpkg.com/en/docs/install
\ No newline at end of file
+[yarn-install]: https://yarnpkg.com/en/docs/install
```
Comment (petebacondarwin) [inline DEVELOPERS.md]: what changed here?

File: TRIAGING.md [hunk 3]
```diff
@@ -95,7 +95,7 @@ You can mention him in the relevant thread like this: `@btford`.
 
 > Thanks for submitting this issue!
 > Unfortunately, we don't think this functionality belongs in core.
-> The good news is that you could easily implement this as a third-party module and publish it on Bower and/or to the npm repository.
+> The good news is that you could easily implement this as a third-party module and publish it to the npm registry.
```
Comment (petebacondarwin) [inline TRIAGING.md]: Do we need this change? It is about other people publishing to bower. Not us using dependencies from bower. Or are you just trying to move everyone away from bower in every possible way?

File: docs/gulpfile.js [hunk 4]
```diff
@@ -13,7 +13,7 @@ var rename = require('gulp-rename');
 
 // We indicate to gulp that tasks are async by returning the stream.
 // Gulp can then wait for the stream to close before starting dependent tasks.
-// See clean and bower for async tasks, and see assets and doc-gen for dependent tasks below
+// See clean for an async task, and see assets and doc-gen for dependent tasks below.
```
Comment (petebacondarwin) [inline docs/gulpfile.js]: Neither `clean` nor `bower` tasks exists in this file. An example of an async task is `doc-gen`.

**Review by petebacondarwin — APPROVED** [review]
A few minor nits but this looks good enough.
I think we should have a README in the `vendor` folder to explain why we have these vendor files, what versions we have here and how to go about updating them.

File: DEVELOPERS.md (hunk 2 above)
Comment (mgol) [reply DEVELOPERS.md]: A new line was added to the end of the file.

File: TRIAGING.md (hunk 3 above)
Comment (mgol) [reply TRIAGING.md]: Yeah, I guess I should revert this change.

Comment (Narretz) [reply TRIAGING.md]: I think it's okay to make this change. bower has no immediate advantages over npm, and is getting less important every week.

**Comment by Narretz** [comment]
Rebased with readme as https://github.com/angular/angular.js/pull/16394

**Pull request closed without merging**